src/shapes/Object/InteractiveObject.ts

```diff
@@ -212,81 +218,81 @@ export class InteractiveFabricObject<
    * @return {Record<string, TOCoord>}
    */
   calcOCoords(): Record<string, TOCoord> {
-    const vpt = this.getViewportTransform(),
-      center = this.getCenterPoint(),
-      tMatrix = createTranslateMatrix(center.x, center.y),
-      rMatrix = createRotateMatrix({
-        angle: this.getTotalAngle() - (!!this.group && this.flipX ? 180 : 0),
-      }),
-      positionMatrix = multiplyTransformMatrices(tMatrix, rMatrix),
-      startMatrix = multiplyTransformMatrices(vpt, positionMatrix),
-      finalMatrix = multiplyTransformMatrices(startMatrix, [
-        1 / vpt[0],
-        0,
-        0,
-        1 / vpt[3],
-        0,
-        0,
-      ]),
-      transformOptions = this.group
-        ? qrDecompose(this.calcTransformMatrix())
-        : undefined,
-      dim = this._calculateCurrentDimensions(transformOptions),
-      coords: Record<string, TOCoord> = {};
+    const vpt = this.getViewportTransform();
+    const center = this.getCenterPoint();
+    const rotation = calcPlaneRotation(
+      multiplyTransformMatrices(vpt, this.calcTransformMatrix())
+    );
+    const angle = radiansToDegrees(rotation);
+
+    // calculate the viewport bbox size of the object by inverting the rotation
+    // in order to measure the bbox according to the object's axis
+    const bboxTransform = multiplyTransformMatrixArray([
+      vpt,
+      createTranslateMatrix(center.x, center.y),
```

Comment:
why a translation is important if all you want to use at the end is width/height?

Reply:
Because we apply the transform on the points with respect to the object center
If we do not translate the rotation will be applied at 0,0